ProcesadorJSV/src/analizadorLexico/AnalizadorLexico.java

```diff
@@ -10,6 +10,10 @@
 
 public class AnalizadorLexico {
 
+    //FIXME language -> should be english
+    //FIXME TOO MUCH STATE

```

Comment:
No entiendo muy bien a qué te refieres con estados y acoplamiento.

Reply:
Acoplamiento es que tus clases se comunican por todo el código, analizadorsintactico necesita el valor de una variable de analizadorlexico y al revés. El acoplamiento es malo porque hace que tengais 1 clase (en 2 ficheros) en vez de 2 clases. Cuando cambieis una tendréis que cambiar la otra. Tendréis más bugs porque modificais el valor de una variable y afecta a la otra clase. Me refiero a esto: https://en.wikipedia.org/wiki/Coupling_(computer_programming)

Cuando hablo de estado es que teneis muchas variables globales (y peor aún, acopladas con otras clases). El estado global es muy malo porque, tiene más bugs, no se puede paralelizar, es más difícil de entender... la mayoría de esas variables globales podrían ser locales con un poco de trabajo. La programación funcional por ejemplo, un paradigma muy de moda, NO tiene variables globales. Yo soy más pragmático, algunas hacen falta, pero no tantas.

https://softwareengineering.stackexchange.com/questions/148108/why-is-global-state-so-evil